lib/act-fluent-logger-rails/logger.rb

```diff
@@ -116,47 +114,55 @@ def add_message(severity, message)
         end
 
       if message.encoding == Encoding::UTF_8
-        @messages << message
+        logger_messages << message
       else
-        @messages << message.dup.force_encoding(Encoding::UTF_8)
+        logger_messages << message.dup.force_encoding(Encoding::UTF_8)
       end
 
       flush if @flush_immediately
     end
 
     def [](key)
-      @map[key]
+      map[key]
     end
 
     def []=(key, value)
-      @map[key] = value
+      map[key] = value
     end
 
     def flush
-      return if @messages.empty?
+      return if logger_messages.empty?
       messages = if @messages_type == :string
-                   @messages.join("\n")
+                   logger_messages.join("\n")
                  else
-                   @messages
+                   logger_messages
                  end
-      @map[:messages] = messages
-      @map[@severity_key] = format_severity(@severity)
+      map[:messages] = messages
+      map[@severity_key] = format_severity(@severity)
       add_tags
 
-      @fluent_logger.post(@tag, @map)
+      @fluent_logger.post(@tag, map)
       @severity = 0
-      @messages.clear
+      logger_messages.clear
       Thread.current[@tags_thread_key] = nil if @tags_thread_key
-      @map.clear
+      map.clear
     end
 
     def add_tags
-      return unless @tags_thread_key && Thread.current[@tags_thread_key]
+      return unless @tags_thread_key && Thread.current.key?(@tags_thread_key)
       @log_tags.keys.zip(Thread.current[@tags_thread_key]).each do |k, v|
-        @map[k] = v
+        map[k] = v
       end
     end
 
+    def logger_messages
+      Thread.current[:logger_messages] ||= []
+    end
+
+    def map
+      Thread.current[:logger_map] ||= {}
+    end
```

Comment:
For these two, do we want to do something similar to what you did for tags to ensure different loggers use different thread local variables? e.g.

```ruby
def logger_messages
  @messages_thread_key ||= "activesupport_tagged_logging_messages:#{object_id}".freeze
  Thread.current[@messages_thread_key] ||= []
end
```